src/lib/list/list.ts

```diff
@@ -119,6 +119,14 @@ export class MatListItem extends _MatListItemMixinBase implements AfterContentIn
               @Optional() private _navList: MatNavList) {
     super();
     this._isNavList = !!_navList;
+
+    // If no type attributed is specified for <button>, set it to "button".
+    // If a type attribute is already specified, do nothing.
+    let element: HTMLElement = this._getHostElement();
+    if (element.nodeName && element.nodeName.toLowerCase() === 'button'
```

Comment:
You could actually check `tagName`.  Since element is already a type of `HtmlElement`, you shouldn't need to check if a `tagName` is defined since it always will have `tagName` by definition of the `HtmlElement` interface.

Additionally, you can just set the attribute `type` without performing a check beforehand.


```ts
if (element.tagName === 'BUTTON') {
  element.setAttribute('type', element.getAttribute('type') || 'button');
}
```

Reply:
I actually prefer always doing `nodeName` since it's guaranteed to be safe. It should probably be in the coding standards...